contents/docs/plugins/build/tutorial.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Tutorial
+sidebar: Docs
+showTitle: true
+---
+
+This tutorial explains the development workflow and best practices, using a hello world plugin. We go from zero to publishing your plugin in the official PostHog repository.
+
+## Pre-Requisites
+
+1. A self-hosted PostHog instance (or a local development environment)
+1. Knowledge of JavaScript (or TypeScript)
+
+## The Plugin
+
+Every plugin begins with either the PostHog Plugin Source Editor, or a new GitHub repository. In both cases, our plugin source code will look like this:
+
+```js
+/* Runs on every event */
+export function processEvent(event, meta) {
+    // Some events (like $identify) don't have properties
+    if (event.properties) {
+        event.properties['hello'] = `Hello ${meta.config.name || 'world'}`
+    }
+    // Return the event to ingest, return nothing to discard  
+    return event
+}
+```
+
+and our config would look like:
+
+```js
+[
+  {
+    "key": "name", // name of key to be accessed using meta. Check value using `meta.config.name`
+    "name": "Person to greet",
+    "type": "string",
+    "hint": "Used to personalise the property `hello`",
+    "default": "",
+    "required": false
+  }
+]
+```
+
+For information on what code to write and what special functions to use, check out [the overview](./overview) and [the developer reference](./reference).
+### Using the Plugin Source Editor
+
+Go to Plugins -> Advanced Tab -> Plugin Editor -> Start Coding
+
+![Plugin Editor Location](../../../images/plugins/plugin-editor-location.png)
+
+Then, click on "Edit Source", and you're good to go. Copy your code and config into the editor, and you're ready to [test the plugin.](#testing)
+
+### Using a GitHub Repository
+
+We have a [GitHub Template](https://github.com/PostHog/posthog-plugin-starter-kit/generate) which helps you create a new repository with all the right files. There are only two files which make up the entire plugin: the `index.js` and `plugin.json`. Your code goes into `index.js`, and your configuration goes into `plugin.json`.
+
+Other than this, there's the `index.test.js` file for tests, and `package.json` for package dependencies and metadata.
+
+Remember to update `package.json` with the appropriate metadata, like name, description, and maintainer.
+
+Once you've written the code in this new repository, you can run it by installing it locally in PostHog. [See testing for more information.](#testing)
+
+#### Plugin Naming Conventions
+
+When creating your repository, follow the naming convention of `posthog-<plugin-name>-plugin`. For example, the hello world plugin repository would be called `posthog-hello-world-plugin`.
```

Comment:
I wonder how people will feel given a lot of our plugins don't follow this

Reply:
Should be fine still, just wondering

Reply:
Precisely why I want to have this section, so that atleast starting from now all plugins follow this convention